Mercury-Store/Mercury-Store/CategoryTabScreen/ViewModels/ProductDetailsViewModel.swift

```diff
@@ -34,15 +35,22 @@ final class ProductsDetailViewModel: ProductsDetailViewModelType {
     func sendImagesToCollection() {
         productImagesSubject.onNext(product.images)
     }
-    func saveToFavourite() {
-        coreDataShared.insertFavouriteProduct(product: SavedProductItem(productID: Decimal(product.id), productTitle: product.title, productImage: product.image.src , productPrice: Double(product.variants[0].price) ?? 0 , productQTY: 0 , producrState: productStates.favourite.rawValue))
+    
+    func toggleFavourite()->Bool {
+        return coreDataShared.toggleFavourite(product: SavedProductItem(
```

Comment:
`toggleFavorite` is supposed to send product id 

You should handle asynchronous cases. For example, you should disable interaction to avoid multiple user clicks or you should update the view with the current state.